### PR TITLE
Use OpenTelemetry SDK in HotROD 

### PR DESCRIPTION
**Pull request by rbroggi**
* replacing the jaeger-client-go ot tracer implementation with the ot-otel bridge implementation

Signed-off-by: rbroggi <ro_broggi@hotmail.com>

## Which problem is this PR solving?
- Issue #3380

## Short description of the changes
- First iteration in replacing the jaeger client ot tracer implementation with the otel bridge ot implementation

### CODE REVIEW

File: examples/hotrod/pkg/tracing/init.go [hunk 1]
```diff
@@ -16,56 +16,39 @@
...
-// Init creates a new instance of Jaeger tracer.
-func Init(serviceName string, metricsFactory metrics.Factory, logger log.Factory) opentracing.Tracer {
-	cfg := &config.Configuration{
-		Sampler: &config.SamplerConfig{},
-	}
-	cfg.ServiceName = serviceName
-	cfg.Sampler.Type = "const"
-	cfg.Sampler.Param = 1
+// Init creates a new bridge opentrace tracer that sends spans to Jaeger. The
+// bridge tracer is bind to an open telemetry tracer
```
Comment (yurishkuro) [inline examples/hotrod/pkg/tracing/init.go:32]:
```suggestion
// Init initializes OpenTelemetry SDK and uses OTel-OpenTracing Bridge
// to return an OpenTracing-compatible tracer.
```

**Review by yurishkuro — COMMENTED** [review]
jaeger-ui submodule update should not be in this PR, please make sure to rebase off master

**Comment by rbroggi** [comment]
Some comments:

With this change something is not really going as planned as the DAG I get is:

image

instead of the one in the [article](https://medium.com/opentracing/take-opentracing-for-a-hotrod-ride-f6e3141f7941) 

Another change that will have to be documented is that we also need to open the grpc port while running the all-in-one container for the go binary to reach jaeger:

```bash
docker run -d -p6831:6831/udp -p16686:16686 -p14268:14268 jaegertracing/all-in-one:latest
```

Will there be a way to update the medium article (it was very useful to me, it would be a shame to loose it)

**Comment by rbroggi** [comment]
The frontend tracers are not connecting to the rest:

image

**Comment by rbroggi** [comment]
> The frontend tracers are not connecting to the rest:
> 
> image

Dear @yurishkuro, would you have any advice on how I could troubleshoot this problem efficiently?

By investigating the code I fail to understand why this change had an impact only in the `frontend` component of the HOTRod. Well I'm not even sure whether the problem lies 100% on the frontend, the only real info I have is that there is a discontinuity between the trace from frontend and the others.

**Comment by yurishkuro** [comment]
I suspect this has something to do with the propagator of the context on the wire. What happens when the code calls tracer.Inject() and the tracer is the one from the bridge?

**Comment by yurishkuro** [comment]
This was finished in #4187

**Pull request closed without merging**